src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -672,10 +672,16 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
      *
      * @param ContainerInterface $container The service container
      *
+     * @throws \InvalidArgumentException When the container is not an instance of ContainerBuilder
+     *
      * @return DelegatingLoader The loader
      */
     protected function getContainerLoader(ContainerInterface $container)
     {
+        if (!$container instanceof ContainerBuilder) {
+            throw new \InvalidArgumentException(sprintf('The container must be an instance of Symfony\Component\DependencyInjection\ContainerBuilder, got %s.', get_class($container)));
```

Comment:
It's also about performance and not checking types ourselves. It crashed before, it will throw an exception, and anyway, use cases where you really pass a non-ContainerBuilder are pretty small. So, I really think this is not worth it. At the minimum, why not just change the phpdoc?